src/main/java/subway/command/CommandRepository.java

```diff
@@ -0,0 +1,40 @@
+package subway.command;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+public class CommandRepository {
+    private final List<Command> commands = new ArrayList<>();
+
+    public Command getCommand(Character key) throws IllegalArgumentException {
+        Optional<Command> foundCommand = this.commands().stream()
+            .filter(command -> command.getKey().equals(key)).findFirst();
+        if (!foundCommand.isPresent()) {
+            throw new IllegalArgumentException("선택할 수 없는 기능입니다.");
+        }
+        return foundCommand.get();
+    }
+
+    public Command createCommand(Character key, String label, Runnable action) {
+        return new Command(this, key, label, action);
+    }
+
+    public List<Command> commands() {
+        return Collections.unmodifiableList(commands);
+    }
+
+    protected void addCommand(Command command) {
+        commands.add(command);
+    }
+
+    protected void deleteCommand(Command command) {
+        commands.remove(command);
+    }
+
+    protected boolean hasCommand(Command command) {
+        return commands.stream().map(Command::getKey).collect(Collectors.toList()).contains(command.getKey());
```

Comment:
스트림을 쓰실때는 한줄에 모두 쓰지 마시고 각 연산마다 엔터를 한번씩 눌러주세요~ 가독성이 훨씬 좋아집니다 :) 아래 코드처럼 작성하시면 됩니다! 

```suggestion
        return commands.stream()
            .map(Command::getKey)
            .collect(Collectors.toList())
            .contains(command.getKey());
```
그리고 이 부분은 코드를 훨씬 짧고 간결하게 작성할 수 있는 방법이 있는데 대신 약간 사전 작업이 필요합니다. 이건 월요일에 같이 해보죠~